### PR TITLE
Decision: API documentation

### PR DESCRIPTION
**Pull request by Namoshek**
# Purpose

Decision on how to document APIs.

# Checklist

- [X] meta data is updated (e.g. README.md of plugins)

@markus2330 please review my pull request

### CODE REVIEW

File: doc/decisions/api_documentation.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+Additionally to modeling data apart from the API, **API Blueprints** also supports schemata modeling, which is more precise than giving examples for requests and responses.
+
+## Implication
+
+- API descriptions should also follow other conventions like the usage of similar error codes.
+
+## Notes
+
+There are many tools available to use with **API Blueprints**, for example the [CLI tool](https://github.com/apiaryio/apiary-client) of apiary. Other tools can be found on the [official website](https://apiblueprint.org/tools.html) of the standard.
```
Comment (markus2330) [inline doc/decisions/api_documentation.md]: Please also refer to tickets with discussions.

**Comment by markus2330** [comment]
Thank you, very nicely done! @omnidan Can you also take a look?

File: doc/decisions/api_documentation.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# REST API documentation
+
+## Issue
+
+A standard way of describing REST APIs offered by tools and plugins for Elektra is required to ease development and usage for and of these. Because many good standards for describing APIs are out there already, an existing one shall be used.
```
Comment (omnidan) [inline doc/decisions/api_documentation.md]: `s/and/any`?

File: doc/decisions/api_documentation.md (hunk 1 above)
Comment (Namoshek) [inline doc/decisions/api_documentation.md]: I thought this is inappropriate in decision documents, but I'll add them.

**Comment by omnidan** [comment]
Looks good! But maybe the first sentence could be reworded a bit, e.g. `To ease development and usage of Elektra, a standard way of describing REST APIs is required.`

File: doc/decisions/api_documentation.md (hunk 2 above)
Comment (Namoshek) [inline doc/decisions/api_documentation.md]: I'll change the formulation to something more clear. ^^

**Pull request merged**